### PR TITLE
Keep feature highlight view implementation in private with limited public properties

### PR DESCRIPTION
**Pull request by jgunaratne**
Keep feature highlight view implementation in private with limited public properties

### CODE REVIEW

**Review by samnm — APPROVED** [review]
LGTM but I'm curious to hear Ian's thoughts.

Having to leave the properties in the public header is a teensy bit imperfect IMO.

File: components/FeatureHighlight/src/MDCFeatureHighlightView.h [hunk 1]
```diff
@@ -14,27 +14,17 @@
...
 typedef void (^MDCFeatureHighlightInteractionBlock)(BOOL accepted);
 
 @interface MDCFeatureHighlightView : UIView
 
-@property(nonatomic, strong) UIColor *outerHighlightColor;
-@property(nonatomic, strong) UIColor *innerHighlightColor;
+@property(nonatomic, strong) UIColor *outerHighlightColor UI_APPEARANCE_SELECTOR;
+@property(nonatomic, strong) UIColor *innerHighlightColor UI_APPEARANCE_SELECTOR;
 
 @property(nonatomic, assign) CGPoint highlightPoint;
```
Comment (ianegordon) [inline components/FeatureHighlight/src/MDCFeatureHighlightView.h]: Please move all non-UI_APPEARANCE_SELECTOR properties to the private header.

**Review by ianegordon — DISMISSED** [review]
I was able to have a public category with a private header, but it required manual visibility manipulation and did not survive pod update.  Your proposed structure is the only one that I have found to work.

File: components/FeatureHighlight/src/MDCFeatureHighlightView.h (hunk 1 above)
Comment (jgunaratne) [reply components/FeatureHighlight/src/MDCFeatureHighlightView.h]: Done. Moved all non-appearance properties into private.

**Comment by jgunaratne** [comment]
Moved non-appearance properties into private.

File: components/FeatureHighlight/src/MDCFeatureHighlightViewController.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- Copyright 2015-present the Material Components for iOS authors. All Rights Reserved.
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (ajsecord) [inline components/FeatureHighlight/src/MDCFeatureHighlightViewController.h]: Please don't change copyright years—the first year is the year when the file was created.

**Review by ajsecord — DISMISSED** [review]

File: components/FeatureHighlight/src/MDCFeatureHighlightViewController.h (hunk 2 above)
Comment (jgunaratne) [reply components/FeatureHighlight/src/MDCFeatureHighlightViewController.h]: Done. Reverted year change.

**Pull request merged**